x/oracle/keeper/task.go

```diff
@@ -134,11 +134,38 @@ func (k Keeper) DeleteClosingTaskIDs(ctx sdk.Context) {
 func (k Keeper) CreateTask(ctx sdk.Context, creator sdk.AccAddress, task types.TaskI) error {
 	savedTask, err := k.GetTask(ctx, task.GetID())
 	if err == nil {
-		if savedTask.IsValid(ctx) {
-			return types.ErrTaskNotClosed
-		}
-		if err := k.DeleteTask(ctx, savedTask); err != nil {
-			return err
+		if _, ok := task.(*types.Task); ok {
+			if savedTask.IsValid(ctx) {
+				return types.ErrTaskNotClosed
+			}
+			if err := k.DeleteTask(ctx, savedTask); err != nil {
+				return err
+			}
+		} else if reqTask, ok := task.(*types.TxTask); ok {
+			oldTask, ok := savedTask.(*types.TxTask)
+			if !ok {
+				return types.ErrInvalidTask
+			}
+			if !reqTask.Equal(oldTask) {
+				return types.ErrInvalidTask
+			}
```

Comment:
I think we don't care whether oldTask is TxTask or not, and whether oldTask equals reqTask. 
The logic could be just (as line 138~143) :
if oldTask is valid, return error
if oldTask is not valid, delete it.

And reqTask should not be equal to oldTask.